hpx/parallel/algorithms/remove_copy.hpp

```diff
@@ -0,0 +1,289 @@
+    ///           element in the destination range, one past the last element
+    ///           copied.
+    ///
+    template <typename ExPolicy, typename InIter, typename OutIter, typename T>
+    typename boost::enable_if<
+        is_execution_policy<ExPolicy>,
+        typename detail::algorithm_result<ExPolicy, OutIter>::type
+    >::type
+    remove_copy(ExPolicy && policy, InIter first, InIter last, OutIter dest,
+        const T& val)
+    {
+        typedef typename std::iterator_traits<InIter>::iterator_category
+            input_iterator_category;
+        typedef typename std::iterator_traits<OutIter>::iterator_category
+            output_iterator_category;
+
+        BOOST_STATIC_ASSERT_MSG(
+            (boost::is_base_of<
+                std::input_iterator_tag, input_iterator_category>::value),
+            "Required at least input iterator.");
+
+        BOOST_STATIC_ASSERT_MSG(
+            (boost::mpl::or_<
+                boost::is_base_of<
+                    std::forward_iterator_tag, output_iterator_category>,
+                boost::is_same<
+                    std::output_iterator_tag, output_iterator_category>
+            >::value),
+            "Requires at least output iterator.");
+
+        typedef typename boost::mpl::or_<
+            is_sequential_execution_policy<ExPolicy>,
+            boost::is_same<std::input_iterator_tag, input_iterator_category>,
+            boost::is_same<std::output_iterator_tag, output_iterator_category>
+        >::type is_seq;
+
+        return detail::remove_copy<OutIter>().call(
+            std::forward<ExPolicy>(policy), is_seq(),
+            first, last, dest, val);
+    }
+
+    /////////////////////////////////////////////////////////////////////////////
+    // remove_copy_if
+    namespace detail
+    {
+        /// \cond NOINTERNAL
+        template <typename Iter>
+        struct remove_copy_if : public detail::algorithm<remove_copy_if<Iter>, Iter>
+        {
+            remove_copy_if()
+              : remove_copy_if::algorithm("remove_copy_if")
+            {}
+
+            template <typename ExPolicy, typename InIter, typename F>
+            static Iter
+            sequential(ExPolicy const&, InIter first, InIter last,
+                Iter dest, F && f)
+            {
+                return std::remove_copy_if(first, last, dest,
+                    std::forward<F>(f));
+            }
+
+            template <typename ExPolicy, typename FwdIter, typename F>
+            static typename detail::algorithm_result<ExPolicy, Iter>::type
+            parallel(ExPolicy const& policy, FwdIter first, FwdIter last,
+                Iter dest, F && f)
+            {
+                typedef typename std::iterator_traits<FwdIter>::value_type T;
+                return copy_if<Iter>().call(
+                    policy, boost::mpl::false_(), first, last, dest,
+                    [&f](T a){ return !f(a); });
+            }
```

Comment:
Same as above, shouldn't this be `[f](T const& a) { ... }` instead? (later, once we have full C++14, this could be `[f = std::move(f)](T const& a) { ... }`)